core/matching/matchers/xml_templated_match.go

```diff
@@ -0,0 +1,102 @@
+package matchers
+
+import (
+	"regexp"
+
+	"github.com/beevik/etree"
+)
+
+var XmlTemplated = "xmltemplated"
+
+var ignoreExpr = regexp.MustCompile("^\\s*\\{\\{\\s*ignore\\s*\\}\\}\\s*$")
+var regExpr = regexp.MustCompile("^\\s*\\{\\{\\s*regex:(.*)\\}\\}\\s*$")
+
+func XmlTemplatedMatch(match interface{}, toMatch string) bool {
+	matchString, ok := match.(string)
+	if !ok {
+		return false
+	}
+
+	// parse xml in mock data into dom tree
+	expected := etree.NewDocument()
+	if err := expected.ReadFromString(matchString); err != nil {
+		return false
+	}
+
+	// parse xml in actual request body into dom tree
+	actual := etree.NewDocument()
+	if err := actual.ReadFromString(toMatch); err != nil {
+		return false
+	}
+
+	// tree matching
+	return compareTree(expected.Root(), actual.Root())
+}
+
+func compareTree(expected *etree.Element, actual *etree.Element) bool {
+	// compare constructure
+	// step 1. compare tag name
+	if expected.Tag != actual.Tag {
+		return false
+	}
+	// step 2. compare node content
+	// case 1: leaf
+	if isLeaf(expected) {
+		// compare text content
+		return compareValue(expected.Text(), actual.Text())
+	}
+	// case 2: children element matching
+	actual_children := actual.ChildElements()
+	// for each expected tag
```

Comment:
May I suggest having the length check earlier to fail fast? 
```
if len(expected.ChildElements()) != len(actual_children) {
		return false
	}
```

before the for loops

Reply:
The logic is changed if we check the length before for loop.

Reply:
Isn't it true that the two will not match if they have different number of child elements?

Reply:
if I add the check, none of your tests complaint, which makes me think that it's assumed implicitly in your logic. 

I'm suggesting it because it would fail fast without running all the matchings for child elements.

Reply:
As you said, it is OK to add a fast-failing check who has nothing to to with the `if` check below.